DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -417,42 +417,54 @@ Function Upgrade-Package {
     $packageUpgradeOuput | ForEach-Object { Write-Verbose -Message $_ }
 
     # Clear Package Cache
-    Get-ChocoInstalledPackage 'Purge'
+    Get-ChocoInstalledPackage -Purge
 }
 
-function Get-ChocoInstalledPackage ($action) {
+function Get-ChocoInstalledPackage {
+    [CmdletBinding()]
+    param (
+        [switch]$Purge,
+        [switch]$NoCache
+    )
+
     $ChocoInstallLP = Join-Path -Path $env:ChocolateyInstall -ChildPath 'cache'
     if ( -not (Test-Path $ChocoInstallLP)){
         New-Item -Name 'cache' -Path $env:ChocolateyInstall -ItemType Directory | Out-Null
     }
     $ChocoInstallList = Join-Path -Path $ChocoInstallLP -ChildPath 'ChocoInstalled.xml'
 
-    if ($action -eq 'Purge') {
-        Remove-Item $ChocoInstallList -Force
-        $res = $true
-    } else {
+    if ( -not $Purge) {

```

Comment:
The previous code if / else was reversed in these changes. Can we have the `if ($Purge.IsPresent)` at the top so it matches the previous code?